src/shared/firewall-util-nft.c

```diff
@@ -902,6 +904,9 @@ int fw_nftables_add_masquerade(
 
         int r;
 
+        if (!socket_ipv6_is_supported() && af == AF_INET6)
+                return -EOPNOTSUPP;
```

Comment:
`socket_ipv6_is_supported()` caches the result internally, so this order in the check is OK.